cilium-sysdump/utils.py

```diff
@@ -158,35 +201,47 @@ def get_pods_status_iterator_by_labels(label_selector, host_ip_filter,
                         .format(label_selector))
         return
 
-    # kubectl field selector supports listing pods based on a particular
-    # field. However, it doesn't support hostIP field in 1.9.6. Also,
-    # it doesn't support set-based filtering. As a result, we will use
-    # grep based filtering for now. We might want to switch to this
-    # feature in the future. The following filter can be extended by
-    # modifying the following kubectl custom-columns and the associated
-    # grep command.
-    host_ip_filter_cmd = "kubectl get pods --no-headers " \
-        "-o=custom-columns=NAME:.metadata.name," \
-        "HOSTIP:.status.hostIP --all-namespaces | grep -E \"{}\" | " \
-        "awk '{{print $1}}'"
-    host_ip_filter_cmd = host_ip_filter_cmd.format(
-        "|".join(map(str, host_ip_filter)))
-    try:
-        filter_output = subprocess.check_output(
-            host_ip_filter_cmd, shell=True, stderr=subprocess.STDOUT,
+    # Separate out IPs and node names if the input from user is mixed.
+    ip_filter = []
+    name_filter = []
+    for elem in node_filter:
+        if net.is_ipaddress(elem):
+            ip_filter.append(elem)
+        else:
+            name_filter.append(elem)
```

Comment:
nit: a more pythonic way of achieving the same
```suggestion
    ip_filter = [s for s in node_filter if is_ipaddress(s)]
    name_filter = [s for s in node_filter if not is_ipaddress(s)]
```